lib/core/src/sync/mod.rs

```diff
@@ -208,6 +227,124 @@ impl SyncService {
 
         Ok(())
     }
+
+    async fn push(&self) -> Result<()> {
+        let outgoing_details = self.persister.get_sync_outgoing_details()?;
```

Comment:
Perhaps we can have one method: `get_records_to_push` which will return all records with revisions we need to push?